usage-collection/src/main/proto/usage.proto

```diff
@@ -2481,3 +2610,79 @@ message CDPVMDiagnosticsFailureType {
     CLEANUP_FAILURE = 7;
   }
 }
+
+// Generated when a service in CDP has diagnostic event to report.
+message CDPDiagnosticEvent {
+  // The resource CRN.
+  string resourceCrn = 1;
+  // The CDP account ID.
+  string accountId = 2;
+  // The CDP service type the event is generated from.
+  ServiceType.Value serviceType = 3;
+  // The result of the diagnostic.
+  CDPDiagnosticResult.Value result = 4;
+  // The result in more details.
+  string details = 5;
+  // The failure message, if available.
+  string failureMessage = 6;
+}
+
+// The service types. Any new services added to CDP which will have their own
+// billable events must be added here. These values should match the customer
+// visible values used in the CLI.
+//
+// This list below is a subset of the values in the Crn.Service enumeration.
+// The exact value need not match, but it should correspond to some Crn.Service.
+//
+// This should match meteringevents.ServiceType.
+message ServiceType {
+  enum Value {
+    // A value indicating the enum is unset.
+    UNSET = 0;
+    // DataHub
+    DATAHUB = 1;
+    // Machine Learning
+    MACHINE_LEARNING = 2;
+    // Data Warehouse
+    DATA_WAREHOUSE = 3;
+    // Data Engineering
+    DATA_ENGINEERING = 4;
+    // Operational Database
+    OPERATIONAL_DB = 5;
+    // Data Flow
+    DATA_FLOW = 6;
+  }
+}
+
+// The CDP diagnostic result.
+message CDPDiagnosticResult {
+  enum Value {
+    // A value indicating the enum is unset.
+    UNSET = 0;
+    // DBus is not available.
+    DBUS_UNAVAILABLE = 1;
+    // Metering heartbeat generation is failing.
+    METERING_HB_FAILURE = 2;
+  }
+}
+
+// Triggered when a CDP stack patcher either finds an affected stack, successfully patches it, or fails.
+message CDPStackPatchEvent {
+  // Resource CRN of the affected stack.
+  string resourceCrn = 1;
+  // Type of stack patch that affects the referenced stack.
+  string stackPatchType = 2;
+  // Type of the stack patch event.
+  CDPStackPatchEventType.Value eventType = 3;
+  // Optional message with more details about the event.
+  string message = 4;
+}
+
+// Type of stack patch event.
+message CDPStackPatchEventType {
+  enum Value {
+      AFFECTED = 0;
```

Comment:
Not sure if it's intended or not: 0th values are also the default values for enums
If it's not desired an UNSET=0 can be added as default value.

Reply:
Before sending this event, in `StackPatchUsageReporterService.sendUsageReport` there is a check that it should not be null, is that not enough?

Reply:
it's totally enough, it was mainly just a FYI if in the feature the new event is used somewhere else in the code to be aware that if the type field is not explicitly set then it will default to "AFFECTED"

Reply:
Thanks, I have added an USNET value